vscode-dotnet-runtime-library/src/test/unit/DebianDistroTests.test.ts

```diff
@@ -5,30 +5,36 @@
  * ------------------------------------------------------------------------------------------ */
 import * as chai from 'chai';
 import * as os from 'os';
-import { MockCommandExecutor, MockEventStream } from '../mocks/MockObjects';
 import { DebianDistroSDKProvider } from '../../Acquisition/DebianDistroSDKProvider';
 import { DotnetInstallMode } from '../../Acquisition/DotnetInstallMode';
 import { DistroVersionPair, DotnetDistroSupportStatus, LinuxVersionResolver } from '../../Acquisition/LinuxVersionResolver';
-import { getMockAcquisitionContext, getMockUtilityContext } from './TestUtility';
+import { MockCommandExecutor } from '../mocks/MockObjects';
+import { getDistroInfo, getMockAcquisitionContext, getMockUtilityContext } from './TestUtility';
+import { DEBIAN_DISTRO_INFO_KEY } from '../../Acquisition/StringConstants';
 const assert = chai.assert;
 const standardTimeoutTime = 100000;
 
 const mockVersion = '7.0.103';
 const acquisitionContext = getMockAcquisitionContext('sdk', mockVersion);
 const mockExecutor = new MockCommandExecutor(acquisitionContext, getMockUtilityContext());
-const pair: DistroVersionPair = { distro: 'Debian GNU/Linux', version: '12' };
+const pair: DistroVersionPair = { distro: DEBIAN_DISTRO_INFO_KEY, version: '12' };
 const provider: DebianDistroSDKProvider = new DebianDistroSDKProvider(pair, acquisitionContext, getMockUtilityContext(), mockExecutor);
-const shouldRun = os.platform() === 'linux';
 const installType: DotnetInstallMode = 'sdk';
 const noDotnetString = `
 dotnet: command not found
 `
 
+async function shouldRun()
+{
+    const distroInfo = await getDistroInfo(acquisitionContext);
+    return os.platform() === 'linux' && distroInfo.distro === DEBIAN_DISTRO_INFO_KEY;
+}
+
 suite('Debian Distro Logic Unit Tests', () =>
 {
     test('Recommends Correct Version', async () =>
     {
-        if (shouldRun)
+        if (await shouldRun())
```

Comment:
So disabling a bunch of tests on non-Debian? Edit: these are the DebianDistroTests, so that actually makes sense 😆 Could any of them be moved to the linux tests?

'await' in a test feels weird to me. I can't point to a real problem with it; it just makes me concerned about hangs

Reply:
Yeah, each distro has its own tests that can be run locally to make sure it works properly. The linux tests also test the same logic, just not the distro specific logic. There is certainly room for improvement. Mocha as a test framework times out every test with our current options, so it should be ok, but I think that's a good point.

Reply:
Alternatively, could we check shouldRun() once then just access it? You aren't going to start a test run on Debian and finish on Ubuntu